apps/src/templates/sectionProgress/SectionProgressToggle.jsx

```diff
@@ -21,20 +21,45 @@ class SectionProgressToggle extends React.Component {
     setCurrentView: PropTypes.func.isRequired,
   };
 
+  state = {
+    selectedToggle: this.props.currentView,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    // currentView can be set externally, and the component will
+    // still need to update when that happens.
+    if (this.state.selectedToggle !== nextProps.currentView) {
+      this.setState({selectedToggle: nextProps.currentView});
+    }
+  }
+
   onChange = () => {
-    if (this.props.currentView === ViewType.SUMMARY) {
-      this.props.setCurrentView(ViewType.DETAIL);
+    // Display the toggle based on the internal state so that it is
+    // more immediately responsive. Once setting internal state is
+    // complete, then update the redux currentView.
+    // Timeouts forces a render of the local state before dispatching
+    // the action.
+    if (this.state.selectedToggle === ViewType.SUMMARY) {
+      this.setState({selectedToggle: ViewType.DETAIL}, () => {
+        setTimeout(() => {
+          this.props.setCurrentView(ViewType.DETAIL);
+        }, 0);
+      });
     } else {
-      this.props.setCurrentView(ViewType.SUMMARY);
+      this.setState({selectedToggle: ViewType.SUMMARY}, () => {
+        setTimeout(() => {
+          this.props.setCurrentView(ViewType.SUMMARY);
+        }, 0);
```

Comment:
Let's deduplicate a bit here - figure out `nextViewType` and then write the setState / setTimeout stuff just once.